.azure-pipelines.yml

```diff
@@ -0,0 +1,40 @@
+#
+# Azure Pipelines configuration for building and testing react on Windows.
+#
+
+trigger:
+- master
+
+jobs:
+- job: Windows
+  pool:
+    vmImage: 'vs2017-win2016'
+  steps:
+  - script: git config --global core.autocrlf input
+    displayName: 'Fix line endings in Windows before cloning'
+  - checkout: self
+  # need to check out master to get merge-base between HEAD and master
+  - script: git checkout -t origin/master
```

Comment:
Isn't this going to change what's in your local source directory? :)

Reply:
Ugh. You're right. I can make it work by creating a local tracking branch for `master` and then checking out the previous HEAD/merge commit. Is that too clunky? I hesitate to change the test itself to `git merge-base HEAD origin/master` since that changes the meaning of the test.

Reply:
Did my earlier suggestion of running `git fetch origin master:master` (anytime after the checkout step) not work? If it would work, maybe think about having a single "post-checkout" script step that does this and sets the core.autocrlf (which, I'm 94% sure can be done after clone, as long as you do a git reset):

```yaml
- script: |
    git config core.autocrlf input
    git reset head --hard
    git fetch origin master:master
  displayName: ...
```

(no need for `checkout: self`)

Reply:
Ah, I think that should work. I think I accidentally tried `git fetch origin master` instead of `git fetch origin master:master`. (Facepalm.)